compiler/codegen/cg-type.cpp

```diff
@@ -187,6 +205,217 @@ void EnumType::codegenDef() {
+    }
+
+    llvm::StructType * st;
+    // handle an empty union.
+    if( largestType ) {
+      st = llvm::StructType::get(largestType);
+    } else {
+      st = llvm::StructType::get(info->module->getContext());
+    }
+
+    const char* unionName = "_u";
+    if (explicitPadding)
+      addFieldWhenEP(unionName, st, largestSize,
+                     (int)DL.getABITypeAlign(st).value());
+    else
+      addFieldWhenNotEP(unionName, st);
+
+  } else if (ag->symbol->hasFlag(FLAG_OBJECT_CLASS)) {
+    //// the RootClass ////
+    llvm::Type* cidType = info->lvt->getType("chpl__class_id");
+    INT_ASSERT(cidType);
+    const char* cidName = "chpl__cid";
+    if (explicitPadding)
+      addFieldWhenEP(cidName, cidType, DL.getTypeStoreSize(cidType),
+                     (int)DL.getABITypeAlign(cidType).value());
+    else
+      addFieldWhenNotEP(cidName, cidType);
+
+  } else {
+    //// a record or class ////
+
+    for_fields(field, ag) {
+      if (field->type == dtNothing || field->type == dtVoid) continue;
+
+      TypeSymbol* fieldSym = field->type->symbol;
+
+      llvm::Type* fieldType = field->hasFlag(FLAG_SUPER_CLASS) ?
+        fieldSym->getLLVMStructureType() : fieldSym->getLLVMType();
+
+      int fieldAlign = field->hasFlag(FLAG_SUPER_CLASS) ?
+        fieldSym->getLLVMStructureAlignment() : fieldSym->getLLVMAlignment();
+
+      if (! explicitPadding &&
+          // skip getABITypeAlign() in the common case of deferred fieldAlign
+          ! isDeferredAlignment(fieldAlign) &&
+          fieldAlign > (int)DL.getABITypeAlign(fieldType).value())
+      {
+        // We had relied on LLVM to align our fields up until this field,
+        // which LLVM may not align correctly. Restart with explicit padding.
+        params.clear(); aligns.clear(); GEPMap.clear();
+        buildStructFields(DL, ag, true, GEPMap, params, aligns,
+                          structSize, structAlignment);
+        return;
+      }
+
+      if (explicitPadding) {
+        fieldAlign = fieldSym->getABIAlignment(fieldType);
+        // add padding before the field if needed
+        if (currSize % fieldAlign > 0)
+          insertPadding(fieldAlign - currSize % fieldAlign);
+      }
+
+      if (explicitPadding)
+        addFieldWhenEP(field->cname, fieldType,
+                       DL.getTypeStoreSize(fieldType), fieldAlign);
+      else
+        addFieldWhenNotEP(field->cname, fieldType);
+
+    } // for_fields
+
+    if (params.empty()) {
+      // add dummyFieldToAvoidWarning which also
+      // stops us from doing 0-byte memory allocation
+      // (comes up with ioNewline and --no-local RA)
+      // TODO - don't ever allocate 0-byte structures
+      // example: DefaultComparator
+      if (explicitPadding)
+        addFieldWhenEP("_dummyFieldToAvoidWarning", padTy, 1, 1);
+      else
+        addFieldWhenNotEP("_dummyFieldToAvoidWarning",
+          llvm::Type::getInt8Ty(gContext->llvmContext())); // here padTy==0
```

Comment:
The old code used a dummy value of i32, this is i8. Why the change?

Reply:
No real good reason other than saving space, at least "on paper". If you feel strongly I can revert to i32.

Reply:
I would revert to i32 to avoid possible unintended side effects